src/mlpack/core/math/mat_redef.hpp

```diff
@@ -0,0 +1,53 @@
+/**
+ * @file core/math/mat_redef.hpp
+ *
+ * A shim around arma and coot functions to avoid confusion with standard 
+ * library functions. This is necessary as we are using ADL to allow the
+ * compiler to deduce to which library functions belings without the need
+ * for namespace. This is mostly needed for MSVC compiiler, gcc seems to
+ * pass without an issue.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ *
+ * Author: Omar Shrit
+ * Author: Ryan Curtin
```

Comment:
I know it is kind of pedantic but could you use the `@author` annotations above the license like the other files?  I have some scripts that check for licenses and I think they can handle the license not being the last thing in the header comment... but I am not 100% sure 😄